services/worker/src/worker/main.py

```diff
@@ -38,11 +50,14 @@ def _create_worker_loop_executor(self) -> OutputExecutor:
     def start(self) -> None:
         worker_loop_executor = self._create_worker_loop_executor()
         worker_loop_executor.start()  # blocking until the banner is printed
+
+        loop = asyncio.get_event_loop()
         logging.info("Starting heartbeat.")
-        while worker_loop_executor.running():
-            self.heartbeat()
-            time.sleep(self.app_config.worker.heartbeat_time_interval_seconds)
-        worker_loop_executor.stop()
+        loop.create_task(every(self.heartbeat, seconds=self.app_config.worker.heartbeat_time_interval_seconds))
+        loop.create_task(every(self.kill_zombies, seconds=self.app_config.worker.kill_zombies_time_interval_seconds))
+        loop.run_until_complete(
+            every(self.is_worker_alive, worker_loop_executor=worker_loop_executor, seconds=1, stop_on=False)
```

Comment:
Do we need this hard-coded? seconds=1

Reply:
We want the executor to stop as soon as the worker stopped running, so it's fine to hardcode it IMO